### ISSUE TITLE
Model Implementation

### ISSUE THREAD

**Issue by ruthvika-raj** [ruthvika-raj/nanogpt#2]
We build a Generatively Pretrained Transformer (GPT), following the paper "Attention is All
You Need".

To build the GPT we use
Transformer Architecture( Token Embeddings, Positional Encoding, Multi-Head SelfAttention, Feed-Forward Networks (MLPs), Causal Masking, Layer Normalization &
Dropout.
We use efficiency techniques : Gradient Accumulation, CUDA & Mixed Precision Training
